### PR TITLE
Updated MDS to 5.0

### PR DESCRIPTION
**Pull request by nemanja-milovancevic**
Update Microsoft.Data.SqlClient to 5.0.0. I built and tested all changed on my dev machine.

I'll replace DacFx version once it's updated to MDS 5.0 too.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
@nemanja-milovancevic Is this required for something you're working on?

@cheenamalhotra Please help validate and make sure this isn't going to cause any problems - 2 major versions is a lot and we've been burned by SqlClient updates in the past so we need to be careful here.

**Comment by Charles-Gagnon** [comment]
(and note that the checklist above should 100% be done before this gets merged in, especially the cross platform testing with all auth types. We've had breaks in a number of different situations before so just testing on one platform for example isn't enough)

File: src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 	<PropertyGroup>
 		<!-- Targeting netstandard2.0 so that other things such as PS cmdlets can use this which need to support a wider range of machines -->
-		<TargetFramework>netstandard2.0</TargetFramework>
+		<TargetFramework>net6.0</TargetFramework>
```
Comment (cheenamalhotra) [inline src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj:4]: Is this a required change?

File: src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (cheenamalhotra) [inline src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj:1]: I hope the msbuild.binlog would be removed?

File: src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs [hunk 3]
```diff
@@ -115,7 +116,8 @@ internal IConnectedBindingQueue ConnectionQueue
 
         static ConnectionService()
         {
-            SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider = new SqlColumnEncryptionAzureKeyVaultProvider(AzureActiveDirectoryAuthenticationCallback);
+            var tokenCredential = new DefaultAzureCredential();
+            SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider = new SqlColumnEncryptionAzureKeyVaultProvider(tokenCredential);
```
Comment (cheenamalhotra) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs:120]: How was the user authenticated before in callback? I could not find any bearer token being attached to the Security token request, can you clarify? Just want to make sure this is a safe replacement.

File: src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs [hunk 4]
```diff
@@ -115,7 +116,8 @@ internal IConnectedBindingQueue ConnectionQueue
 
         static ConnectionService()
         {
-            SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider = new SqlColumnEncryptionAzureKeyVaultProvider(AzureActiveDirectoryAuthenticationCallback);
```
Comment (cheenamalhotra) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs:118]: Please remove callback if not needed anymore.

**Comment by nemanja-milovancevic** [comment]
> @nemanja-milovancevic Is this required for something you're working on?
> 
> @cheenamalhotra Please help validate and make sure this isn't going to cause any problems - 2 major versions is a lot and we've been burned by SqlClient updates in the past so we need to be careful here.

**Comment by nemanja-milovancevic** [comment]
> @nemanja-milovancevic Is this required for something you're working on?
@Charles-Gagnon Yes, I'm working on updating ADS to TDS 8.0.

**Comment by nemanja-milovancevic** [comment]
I'll close the PR now since @cheenamalhotra is continuing work on this in the https://github.com/microsoft/sqltoolsservice/pull/1708.

**Pull request closed without merging**